### PR TITLE
Add proposal for `Implicit Default Optional Parameters`.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Proposal Issue #29

### CODE REVIEW

File: proposals/proposal-Implicit-default-optional-parameters.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+   DefaultToValue ::= "=" ( "Nothing" | ConstantValue )
+```
+
+**Examples**
+
+```vb.net
+Foo( Optional arg0 )                     ' --> Foo( Optional arg0 As Object = Nothing )
+Foo( Optional arg1 As String )           ' --> Foo( Optional arg1 As String = Nothing )
+Foo( Optional arg2 As String = Nothing ) ' --> Foo( Optional arg2 As String = Nothing )
+Foo( Optional arg3 = "" )                ' --> Foo( Optional arg3 As String = "" )
```
Comment (AnthonyDGreen) [inline proposals/proposal-Implicit-default-optional-parameters.md]: This is a breaking change. I believe today this means Optional arg3 As Object = "".

File: proposals/proposal-Implicit-default-optional-parameters.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Foo( Optional arg4 As Integer )          ' --> Foo( Optional arg4 As Integer = Nothing )
+Foo( Optional arg5 As Integer = 0 )      ' --> Foo( Optional arg5 As Integer = 0 )
+Foo( Optional arg6 As Integer = 1 )      ' --> Foo( Optional arg6 As Integer = 1 )
+Foo( Optional arg7 = 7 )                 ' --> Foo( Optional arg7 As Integer = 7 )
```
Comment (AnthonyDGreen) [inline proposals/proposal-Implicit-default-optional-parameters.md]: This is a breaking change. I believe today this means Optional arg7 As Object = 7.

**Comment by AdamSpeight2008** [comment]
@AnthonyDGreen I've updated the proposal.

**Comment by AdamSpeight2008** [comment]
@AnthonyDGreen  The prototype has been updated.

**Comment by AdamSpeight2008** [comment]
@AnthonyDGreen  The prototype is ready for a review.

**Pull request merged**